### PR TITLE
Get dashboard ready for AWS dev deployment

### PR DESCRIPTION
**Pull request by kparlante**
- Added AWS configuration
- Updated update script to use https and date arguments
- Added persona login

I've made most of the recommended changes, except:
- Didn't move styles to CSS yet
- Config needs to be handled better #14. A particular ugliness is that you have to ssh in to your ephemeral instance and edit the audience config.

### CODE REVIEW

File: server/bin/app [hunk 1]
```diff
@@ -3,12 +3,89 @@
...
+    app = express();
 
+app.set('views', __dirname + '/../views');
+app.set('view engine', 'ejs');
 app.use(express.logger());
 app.use(express.static(__dirname + '/../../static'));
 
+app.use(express.bodyParser());
+app.use(express.cookieParser());
+app.use(express.session({ secret: 'dashkpidashkpidashkpi'}));
```
Comment (shane-tomlinson) [inline server/bin/app:19]: Is it alright for this secret to be embedded here? I've always wondered that.

File: server/config/config.json [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+        "port": 443,
         "path": "/wsapi/interaction_data"
     },
     "database_server": {
         "host": "localhost",
         "port": 5984,
-        "database": "kpi_dashboard"
+        "database": "kpi_dashboard",
+        "username": "dashboard",
+        "password": "dashboard"
```
Comment (shane-tomlinson) [inline server/config/config.json:12]: Is this alright to have here?

File: server/lib/data.js [hunk 3]
```diff
@@ -2,53 +2,37 @@
...
-    });
+exports.getData = function(options, callback) {
+  
+  var params = {
+    host: config.data_server.host,
+    port: config.data_server.port,
+    path: config.data_server.path + "?" + querystring.stringify(options)
+  }
+
+  https.get(params, function(res) {
```
Comment (shane-tomlinson) [inline server/lib/data.js:24]: Will this always be https?

File: server/scripts/aws_post_create.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+# This script is not currently used
+# If we could get awsbox to load up an extra script on postcreate we might move the steps to start up the db and create users here
```
Comment (shane-tomlinson) [inline server/scripts/aws_post_create.sh:4]: You could change the .awsbox.json postcreate hook to call this. From here, call the other scrips like couchdb-ec2-install.sh, create-users, and the database startup.

Comment (shane-tomlinson) [inline server/scripts/aws_post_create.sh:4]: Oh, from https://github.com/mozilla/kpi-dashboard/pull/24/files#L9R6, it looks like this is already done. Is this script necessary? If not, can you remove it?

File: server/scripts/couchdb-ec2-install.sh [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+require_valid_user = false
+
+[log]
+level = debug
+
+[admins]
+EOF
+
+# allow beam to bind to port 80 (not necessary if you make httpd.port >=1024)
+sudo setcap 'cap_net_bind_service=+ep' /usr/local/lib/erlang/erts-5.9.2/bin/beam
```
Comment (shane-tomlinson) [inline server/scripts/couchdb-ec2-install.sh:51]: What is beam? Something for couch?

File: server/scripts/couchdb-ec2-install.sh [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+if [ ! -e "/etc/logrotate.d/couchdb" ]
+then
+  # add couch.log to logrotate
+  sudo ln -s /usr/local/etc/logrotate.d/couchdb /etc/logrotate.d/
+  # change to daily rotation
+  sudo sed -e s/weekly/daily/g -i /usr/local/etc/logrotate.d/couchdb
+  #logrotate -v -f /etc/logrotate.d/couchdb
```
Comment (shane-tomlinson) [inline server/scripts/couchdb-ec2-install.sh:59]: Should this be commented out?

File: server/views/dashboard.ejs [hunk 7]
```diff
@@ -54,17 +54,27 @@
...
             <div class="navbar">
                 <div class="navbar-inner">
-                        <h1><a class="brand">KPI Dashboard</a></h1>
+                    <h1><a class="brand">KPI Dashboard</a>
+                        <% if (mozilla_user) { %>
+                        <a id="signout">logout</a>
+                        <% } else { %>
+                        <img id="signin" alt="Sign in"
+                        src="https://developer.mozilla.org/files/3971/plain_sign_in_red.png" 
+                        style="width: 95px; height: 25px;">
```
Comment (shane-tomlinson) [inline server/views/dashboard.ejs:67]: Do you want these styles here or in CSS?

File: server/views/dashboard.ejs [hunk 8]
```diff
@@ -54,17 +54,27 @@
             }
         </style>
     </head>
-    <body>
+    <body data-logged-in-user="<%= persona_user %>">
```
Comment (shane-tomlinson) [inline server/views/dashboard.ejs:57]: I dig it.

File: server/bin/app (hunk 1 above)
Comment (ozten) [inline server/bin/app:19]: No, this must come from the config.

**Comment by shane-tomlinson** [comment]
This is _much_ easier to set up! Fantastic stuff.

When I got to log on in my [ephemeral deployment](https://kdash-shane.personatest.org/), I get stuck in an endless loop. Is this expected?

**Comment by shane-tomlinson** [comment]
> Config needs to be handled better #14. A particular ugliness is that you have to ssh in to your ephemeral instance and edit the audience config.

AHA!

File: server/scripts/aws_post_create.sh (hunk 4 above)
Comment (kparlante) [inline server/scripts/aws_post_create.sh:4]: Awsbox loads and runs the post-create hook script before you've done the first git push, so you don't have other files available. In the interest of getting things done I squashed the files together, but would like to modularize it later.

File: server/scripts/couchdb-ec2-install.sh (hunk 5 above)
Comment (kparlante) [inline server/scripts/couchdb-ec2-install.sh:51]: beam is the couch process (Erlang!). It shouldn't be binding to port 80, so I'll remove the line.

File: server/bin/app (hunk 1 above)
Comment (kparlante) [inline server/bin/app:19]: Good catch, thanks.

File: server/config/config.json (hunk 2 above)
Comment (kparlante) [inline server/config/config.json:12]: When we clean up the configuration (#14), we should pull this and other config entries from the environment, but seems reasonable to have defaults for dev instances here.

File: server/lib/data.js (hunk 3 above)
Comment (kparlante) [inline server/lib/data.js:24]: Yeah, we should probably make this work via http or https in case you wanted to run it against a local kpiggybank.

**Pull request merged**